contracts/DutchExchangeHelper.sol

```diff
@@ -125,7 +123,8 @@ contract DutchExchangeHelper {
 
         uint arrayLength;
 
-        uint startingIndex = lastNAuctions == 0 ? 1 : runningAuctionIndex - lastNAuctions + 1;
+        uint startingIndex = lastNAuctions == 0 || lastNAuctions > runningAuctionIndex
+        ? 1 : runningAuctionIndex - lastNAuctions + 1;
```

Comment:
I would add parenthesis to make more clear the expression, or maybe rewrite it in a simpler way